### PR TITLE
Use `currentcolor` as border-color for outline button style

### PR DESCRIPTION
**Pull request by chrisvanpatten**
This updates the "Outline" button style to use `currentcolor`—aka the current text color—as the border color, per a @mtias [comment in Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1539699415000100?thread_ts=1539697773.000100&cid=C02QB2JS7) (I'm considering that comment "design feedback" but I'm also happy to request additional feedback if desired.)

To override a potential inline `style` attribute, an `!important` is unfortunately required. I think this is acceptable in this case but I'm also open to other ideas!

## Screenshot

edit_page_ _tomodomo_ _wordpress

### CODE REVIEW

**Comment by ZebulanStanphill** [comment]
Why can't the background color option still be used?

**Comment by chrisvanpatten** [comment]
@ZebulanStanphill The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through.

**Comment by ZebulanStanphill** [comment]
@chrisvanpatten But what if I want to have a button with an outline _and_ a background? If you want the background to be transparent, why not just clear the background color in the inspector? It seems weirder to have the background color option still visible but do nothing.

**Comment by chrisvanpatten** [comment]
@ZebulanStanphill If you need a different style, you can register a new block style that provides whatever capabilities you need.

There is also discussion of changing the code to hide the background color (or other options, as appropriate) based on the block style, but this is focussed on the CSS side, not the interface side.

As to why we are putting this limit in place, it's fundamentally a subjective design decision, but with the ability to register additional styles users should have flexibility to build out whatever visuals they need.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Review by jasmussen — COMMENTED** [review]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Thank you for this PR. This is cool, and it seems to work well, both for the placeholder, the style preview, and the color thing itself:

screenshot 2018-10-22 at 10 53 51

There's one thing we need to figure out though, and you're already discussing it. In fact I agree equally with both of you:

> Why can't the background color option still be used?

> The outline style is intended to have a transparent background, so content behind it (such as an image) peeks through. If you want to use a background, you can use the default/normal button style.

The outline style is exactly as noted, intended to be transparent so content behind can peek through. You may already be bored to death of this, but imagine a Cover image dimmed with fixed background, then an outline button on top.

However then there's this:

bg

Basically, the background option in the sidebar is still present, but in this style it simply doesn't do anything. This is confusing, and although the argument about this variation being transparent is right, I don't think a layperson would grok this.

So it seems there are two paths ahead:

1. Let the background color work. Okay, it goes against the spirit of the style to have a background color, but at least it will be functional. Maybe you can make it semi-transparent to sort of embrace the transparent nature?

2. Hide the background color option when this style is picked. Not sure how easy or feasible this is.

What do you think? 1 seems like the easiest fix.

**Comment by jasmussen** [comment]
Okay I don't know what's going on with Github right now, but I've saved my comments as text files. They seem to show up _some times_ and not others.

I apologize if I've spammed you all with emails for trying to submit multiple times, but this outage has been worse than most!

**Comment by ajitbohra** [comment]
@jasmussen seems like an issue on GitHub can see multiple comments here and on other issues/PRs.

**Comment by chrisvanpatten** [comment]
Travis is passing so we're ready for review! 🚀

**Comment by mtias** [comment]
Actually, I think if you have the "outline" style the background color should be reset and the options removed from the inspector.

**Comment by ZebulanStanphill** [comment]
@mtias Personally, I disagree. I think hiding options in the inspector based on the selected style variation is kind of weird and confusing.

**Comment by chrisvanpatten** [comment]
I'm happy either way although right now I don't think we can get style variations in the Inspector to conditionally show/hide fields (since they were intended to just apply simple `class` names).

**Comment by youknowriad** [comment]
@chrisvanpatten Providing the current style as a prop to `edit` functions is something I was planning to work on soon.

**Comment by mtias** [comment]
>  I think hiding options in the inspector based on the selected style variation is kind of weird and confusing.

Why? Some options might only be relevant to certain styles and confusing with others.

**Comment by jasmussen** [comment]
I'll defer to the others, because I'm good with that.

**Review by youknowriad — APPROVED** [review]
Changes look good to me 👍

**Comment by drw158** [comment]
Not sure what was decided on, but I think there's another option. I agree it's a bad experience when the background settings don't appear to be doing anything, but it's also not great when settings appear based on which style is selected. 

Before reading this, I fully expected the background setting to apply to the border of the button (when I have the outlined style applied). 

So instead of this:

screen shot 2018-10-30 at 10 41 55 am

it would be this

screen shot 2018-10-30 at 10 46 03 am

---

You would change the border color of the outlined button by changing the background color. I think both of issues presented earlier would be solved with this method.

Yes, technically "background color" is not the same as "border color" but I wouldn't assume it would matter to users. Background color is just the color of the button, whether it be an outlined button or a solid button. As a user I still consider the outline the "background" of the button — it just a background with a border and a transparent fill. Hope this helps!

**Comment by shaunandrews** [comment]
Instead of "Background Color" maybe "Shape Color" or "Button Color"?

**Comment by aduth** [comment]
@webmandesign You may be right, specifically by the standard behavior of an unset `border-color` defaulting to the element's `color` value:

>The border color properties specify the color of a box's border.
>
>Initial: | the value of the 'color' property

https://www.w3.org/TR/CSS2/box.html#border-color-properties

I guess if we can assume "the value of the 'color' property" and `currentColor` are interchangeable, there's no need for it.

**Comment by chrisvanpatten** [comment]
My only concern is whether we need the extra specificity for theme compat, but if someone can test and verify it works without `currentcolor` with the `twenty{$year}` themes, I see no reason to keep it in!

**Comment by webmandesign** [comment]
@aduth Yes, those are interchangeable for sure. I have a lot of experience with border colors in my themes ;)

@chrisvanpatten This should cause no issue. We already have an override with:
`.wp-block-button.is-style-outline .wp-block-button__link { border: 2px solid; }` That resets all border styles, including color.

I'll go ahead an create an issue ticket about `currentColor` in border declaration and I'll also submit a PR fixing all the issues I've found.

**Comment by webmandesign** [comment]
FYI, #12328
